pkg/asset/k8s.go

```diff
@@ -116,16 +118,38 @@ const validBootstrapTokenChars = "0123456789abcdefghijklmnopqrstuvwxyz"
 // newBootstrapToken constructs a bootstrap token in conformance with the following format:
 // https://kubernetes.io/docs/admin/bootstrap-tokens/#token-format
 func newBootstrapToken() (id string, secret string, err error) {
-	// Read 6 random bytes for the id and 16 random bytes for the token (see spec for details).
-	token := make([]byte, 6+16)
-	if _, err := rand.Read(token); err != nil {
+	idBytes := make([]byte, 6)
+	secretBytes := make([]byte, 16)
+	tokenLength := len(idBytes) + len(secretBytes)
+	charsetLength := len(validBootstrapTokenChars)
+	size := int(math.Ceil(float64(tokenLength) * math.Log2(float64(charsetLength)) / math.Log2(256)))
```

Comment:
Add a comment on what this math is doing. It's not clear why this works.

Reply:
> Add a comment on what this math is doing.

You want `256^size > len(token)^len(charset)`.  Taking the log2 of both sides:

    log2(256^size) > log2(len(token)^len(charset))

bringing the exponents out of the logs:

    size * log2(256) > len(charset)*log2(len(token))

dividing by `log2(256)`:

    size > len(charset)*log2(len(token)) / log2(256)

We can't read a fractional byte, so `Ceil` the whole right hand side to get the smallest integer `size` with enough data, and that's where the right hand side comes from.

I've pushed a version of that explanation into the commit message with abc1878b, where it's available to folks who want to blame the line but not distracting to folks who either understand or don't care.  Let me know if you want me to add the explanation to comments in the file as well.